apps/server/src/modules/authentication/strategy/oauth2.strategy.ts

```diff
@@ -1,33 +1,47 @@
-import { ICurrentUser } from '@infra/auth-guard';
-import { AccountService } from '@modules/account';
-import { OAuthService } from '@modules/oauth';
+import type { ICurrentUser } from '@infra/auth-guard';
+import { Account, AccountService } from '@modules/account';
+import {
+	OAuthService,
+	OauthSessionToken,
+	OauthSessionTokenFactory,
+	OauthSessionTokenService,
+	OAuthTokenDto,
+} from '@modules/oauth';
 import { Injectable } from '@nestjs/common';
 import { PassportStrategy } from '@nestjs/passport';
+import type { UserDO } from '@shared/domain/domainobject';
 import { Strategy } from 'passport-custom';
 import { Oauth2AuthorizationBodyParams } from '../controllers/dto';
 import { StrategyType } from '../interface';
-import { AccountNotFoundLoggableException, SchoolInMigrationLoggableException } from '../loggable';
-import { UserAccountDeactivatedLoggableException } from '../loggable/user-account-deactivated-exception';
+import {
+	AccountNotFoundLoggableException,
+	SchoolInMigrationLoggableException,
+	UserAccountDeactivatedLoggableException,
+} from '../loggable';
 import { CurrentUserMapper } from '../mapper';
 
 @Injectable()
 export class Oauth2Strategy extends PassportStrategy(Strategy, StrategyType.OAUTH2) {
-	constructor(private readonly oauthService: OAuthService, private readonly accountService: AccountService) {
+	constructor(
+		private readonly oauthService: OAuthService,
+		private readonly accountService: AccountService,
+		private readonly oauthSessionTokenService: OauthSessionTokenService
+	) {
 		super();
 	}
 
 	public async validate(request: { body: Oauth2AuthorizationBodyParams }): Promise<ICurrentUser> {
 		const { systemId, redirectUri, code } = request.body;
 
-		const tokenDto = await this.oauthService.authenticateUser(systemId, redirectUri, code);
+		const tokenDto: OAuthTokenDto = await this.oauthService.authenticateUser(systemId, redirectUri, code);
 
-		const user = await this.oauthService.provisionUser(systemId, tokenDto.idToken, tokenDto.accessToken);
+		const user: UserDO | null = await this.oauthService.provisionUser(systemId, tokenDto.idToken, tokenDto.accessToken);
 
 		if (!user || !user.id) {
 			throw new SchoolInMigrationLoggableException();
 		}
 
-		const account = await this.accountService.findByUserId(user.id);
+		const account: Account | null = await this.accountService.findByUserId(user.id);
```

Comment:
why null?

Reply:
because that is a possible return value of the function